pkg/virtual/apiexport/admission/permissionclaim_admission.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"k8s.io/apiserver/pkg/admission"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/kcp-dev/kcp/pkg/permissionclaim"
+	virtualcontext "github.com/kcp-dev/kcp/pkg/virtual/framework/context"
+	kcpinformers "github.com/kcp-dev/kcp/sdk/client/informers/externalversions"
+)
+
+const (
+	PluginName = "apis.view.kcp.io/PermissionClaims"
+)
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(configFile io.Reader) (admission.Interface, error) {
+		return NewPermissionClaimAdmitter(), nil
+	})
+}
+
+type permissionClaimAdmitter struct {
+	*admission.Handler
+
+	apiBindingsHasSynced cache.InformerSynced
+
+	permissionClaimLabeler *permissionclaim.Labeler
+}
+
+var _ admission.ValidationInterface = &permissionClaimAdmitter{}
+var _ admission.InitializationValidator = &permissionClaimAdmitter{}
+
+// NewPermissionClaimAdmitter creates a mutating admission plugin that is responsible for admitting objects
+// according to permission claims. On every creation and update request, we will determine the bindings
+// in the workspace and if the object is claimed by an accepted permission claim we will allow it.
+func NewPermissionClaimAdmitter() *permissionClaimAdmitter {
+	p := &permissionClaimAdmitter{
+		Handler: admission.NewHandler(admission.Create, admission.Update, admission.Delete, admission.Connect),
```

Comment:
@nrb I know you had asked about operations to support here previously and if Connect was needed. Upon further reflection, I think we might want to reduce this just to Create and Update (to handle the create-on-update case), as everything else is covered by labeling and the label selection in the views.

Reply:
👍 make sense.